include/seqan3/alignment/scoring/detail/simd_matrix_scoring_scheme.hpp

```diff
@@ -0,0 +1,150 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::simd_matrix_scoring_scheme.
+ * \author Joshua Kim <joshua.kim AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/configuration/align_config_mode.hpp>
+#include <seqan3/alignment/scoring/scoring_scheme_concept.hpp>
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/core/concept/cereal.hpp>
+#include <seqan3/core/simd/concept.hpp>
+#include <seqan3/core/simd/simd_algorithm.hpp>
+#include <seqan3/std/concepts>
+
+namespace seqan3::detail
+{
+
+/*!\brief A vectorised scoring scheme to handle scoring matricies.
+ * \ingroup scoring
+ * \tparam simd_score_t The type of the simd vector; must model seqan3::detail::simd_concept.
+ * \tparam alphabet_t The type of the alphabet over which to define the scoring scheme; must model seqan3::semialphabet
+ *                    and must have an alphabet size of at least 2.
+ * \tparam alignment_t The type of the alignment to compute; must be either seqan3::detail::global_alignment_type or
+ *                     seqan3::detail::local_alignment_type.
+ * \tparam scoring_scheme_t The type of the scoring scheme which will be used. Must model seqan3::scoring_scheme with
+ *                          the given alphabet_t.
+ *
+ * \details
+ *
+ * Wraps a regular scoring scheme matrix by storing the matrix in a private member variable. Then, when scoring two
+ * seqan3::detail::simd vectors, it performs element-wise lookups into the scoring matrix and returns the result in
+ * another seqan3::detail::simd vector.
+ * Note during the conversion to the simd vectors the alphabet type information is lost and
+ * only the ranks of the alphabet are used.
+ */
+template <simd_concept simd_score_t, semialphabet alphabet_t, typename alignment_t, typename scoring_scheme_t>
+//!\cond
+    requires seqan3::alphabet_size<alphabet_t> > 1 &&
+             scoring_scheme<scoring_scheme_t, alphabet_t> &&
+             (std::same_as<alignment_t, detail::local_alignment_type> ||
+              std::same_as<alignment_t, detail::global_alignment_type>)
+//!\endcond
+class simd_matrix_scoring_scheme
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief Default.
```

Comment:
```suggestion
    //!\brief Defaulted.
```

x6


You can probably also use the inline notation
```
//!< Defaulted.
```